Samples/resource-owner-password-credential/Angular2/Angualar2-Client-ROPC/app/authorize/authorize-component.ts

```diff
@@ -0,0 +1,202 @@
+﻿import {Component, ViewChild } from '@angular/core'
+import {JwtHelper} from 'angular2-jwt'
+import {Router} from '@angular/router';
+import {authervice} from './authoriza-service';
+import { MODAL_DIRECTIVES, ModalComponent  } from 'ng2-bs3-modal/ng2-bs3-modal';
+declare var System;
+@Component({
+    selector: 'authorize',
+    templateUrl: '/app/authorize/authorize-component.html',
+    directives: [MODAL_DIRECTIVES],
+})
+
+export class authorizeComponent{
+
+    constructor(public jwtHelper: JwtHelper,
+        private _parentRouter: Router,
+        private authentication: authervice) {}
+
+    @ViewChild('myModal')
+    modal: ModalComponent;
+
+    public mclose() {
+        this.modal.close();
+    }
+
+   public mopen() {
+        this.modal.open();
+    }
+
+   public logstatus() {
+       this.isLoggedin = true; 
+   }
+
+    public userDetails: any ;
+    public isLoggedin: boolean;
+    public logMsg: string;
+    public model: logModel;
+    public rmodel: registerModel;
+    public login: boolean;
+    public register: boolean;
+    public user: string;
+    public authdata: any;
+    public hodeModel: boolean = false;
+
+    ngOnInit() {
+       
+        var instance = this;
+         // check if auth key is present
+        if (localStorage.getItem('auth_key')) {
```

Comment:
this should be a AuthService responsibility, so consider exposing a method on the service instead